### ISSUE TITLE
Bug: +31970xxxxxxx not allowed for registration

### ISSUE THREAD

**Issue by PieterDeBruijn** [signalapp/Signal-Android#11017]
### Bug Description

In The Netherlands all data-sims (sim cards that offer SMS and 2/3/4/5G internet, but no traditional voice), have been move to a new range. Regular mobile numbers are still in the +316xxxxxxxx range, data sims are in the +3197xxxxxxxx range and have one digit more/are one digit longer.
### Steps to Reproduce

1.  step one: enter the mobile number to activate:
![Screenshot_20210221-195533](https://user-images.githubusercontent.com/3070301/108840074-22122e00-75d6-11eb-8d71-a14b03da93bb.png)
Actual Result: Get error message:
![Screenshot_20210221-195613](https://user-images.githubusercontent.com/3070301/108840069-21799780-75d6-11eb-8905-56048b61e692.png)

Expected Result:
Accept number and send SMS for verification.

The extra digit should be accepted for Dutch numbers.

### Platform Info
Signal Version: All recent versions
Operating System: Android 11

**Comment by hiqua** [comment]
So you cannot receive calls on these numbers right? From the little information I could find, these numbers are meant mostly for automated use: https://www.twilio.com/blog/2016/09/introducing-mobile-numbers-in-the-netherlands.html

Are they now available for regular users?

Right now the validation fails in PhoneNumberUtils.java, from the libphonenumber library, because the number is too long. So that would be a bug to report to them, if these numbers can indeed be used by regular users to receive text messages.

In their FAQ, they have this to say about M2M, but that would not apply to these numbers if your information is correct:
https://github.com/google/libphonenumber/blob/master/FAQ.md#what-about-m2m-machine-to-machine-numbers

In a nutshell, I'd suggest to file a bug on their issue tracker directly. Signal could also whitelist this kind of number though.

**Comment by bme13** [comment]
Hi. This is a duplicate of #9005.

**Comment by PieterDeBruijn** [comment]
Thanks for the help!
